ports/kddockwidgets/portfile.cmake

```
@@ -21,9 +21,14 @@ vcpkg_cmake_configure(
     OPTIONS
         ${_qarg_OPTIONS}
         -DKDDockWidgets_QT6=ON
+        -DKDDockWidgets_FRONTENDS="qtwidgets"

```

```suggestion
        "-DKDDockWidgets_FRONTENDS=qtwidgets"
```
Quote for CMake, not for the shell.

---

Thanks for the comment. After testing, I found CMake does allow keys with spaces, like `-D"a  b=  c  ;d"`.

The `KDDockWidgets_FRONTENDS` key does not contains spaces. Just personally, for variables which name have no spaces, could it add only necessary quotes (like list)?

---

"name" doesn't matter. We look at the "value". (I assume this was meant.)

We have to be careful with: space, quote, semicolon.

- Inner quotes in cmake language are part of the cmake value.
- Outer quotes in cmake language are not part of the value.
- Here, you do not want the quotes to be part of the value.
- **Here, you could just omit the quotes** because there is nothing which needs to be quoted for the *literal value* `qtwidgets`.
- We usually need to quote the options when we do not fully control the value and variables, as in
  ~~~
  "-DPKG_CONFIG_EXECUTABLE=${PKGCONFIG}"
  ~~~
  where the `PKGCONFIG` *variable* might have space or other evil.
- Depending on how values are passed on when calling system commands, possibly through the indirection of a `Makefile`, `build.ninja`, and/or shell, unintented inner quotes may break only some environments.

---

Thank you very much for your explanation. After testing, there are indeed two types of quotes.

And according to the test results, that any arguments passed into function must be fully quoted if it contains special characters, i.e., use outer quotes.

```cmake
cmake_minimum_required(VERSION 3.25.1)

project(TestQuotationMark LANGUAGES NONE)

include(CMakePrintHelpers)

function(TestMultiArgs)
    cmake_parse_arguments(PARSE_ARGV 0 arg "" "" "OPTIONS")
    include(CMakePrintHelpers)

    cmake_print_variables(arg_OPTIONS)

    foreach("                        option" IN LISTS arg_OPTIONS)
        cmake_print_variables("                        option")
    endforeach()

endfunction()

TestMultiArgs(OPTIONS a)
TestMultiArgs(OPTIONS b c)
TestMultiArgs(OPTIONS "d")

# Try to pass a list of values to the OPTIONS argument
TestMultiArgs(OPTIONS "e;f")                # meet the intent
set(some_list "g;h")
TestMultiArgs(OPTIONS ${some_list})         # unexpected
set(some_list "i;j")
TestMultiArgs(OPTIONS "${some_list}")       # meet the intent
set(some_list "k;l")
TestMultiArgs(OPTIONS -Dvar=${some_list})   # unexpected
set(some_list "m;n")
TestMultiArgs(OPTIONS -Dvar="m;n")          # unexpected
set(some_list "o;p")
TestMultiArgs(OPTIONS -Dvar="${some_list}") # unexpected
set(some_list "q;r")
TestMultiArgs(OPTIONS -D"var=${some_list}") # unexpected
set(some_list "s;t")
TestMultiArgs(OPTIONS "-Dvar=${some_list}") # meet the intent
```

Thank you again for your help.